.github/workflows/precommit.yml

```diff
@@ -130,182 +163,15 @@ jobs:
-        run: .github/scripts/decrypt_parameters.sh
-        env:
-          PARAMETER_PASSWORD: ${{ secrets.PARAMETER_PASSWORD }}
-          CLOUD_PROVIDER: ${{ matrix.cloud-provider }}
-      - name: Download wheel(s)
-        uses: actions/download-artifact@v4
-        with:
-          name: wheel
-          path: dist
-      - name: Show wheels downloaded
-        run: ls -lh dist
-        shell: bash
-      - name: Run tests
-        run: ./ci/test_fips_docker.sh
-        env:
-          PYTHON_VERSION: 3.8
-          cloud_provider: ${{ matrix.cloud-provider }}
-          PYTEST_ADDOPTS: --color=yes --tb=short
-          TOX_PARALLEL_NO_SPINNER: 1
-        shell: bash
-      - uses: actions/upload-artifact@v4
-        with:
-          name: coverage_linux-fips-3.8-${{ matrix.cloud-provider }}
-          path: |
-            .coverage
-            coverage.xml
-
-  test-disable-sql-simplifier: # Will be removed after sql simplifier is stable and no option to opt out.
-    name: Test Disable SQL Simplifier py-${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
-    needs: build
-    runs-on: ${{ matrix.os.image_name }}
-    strategy:
-      fail-fast: false
-      matrix:
-        os:
-          - image_name: macos-latest
-            download_name: macos  # it includes doctest
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
-        cloud-provider: [aws]
-    steps:
-      - name: Checkout Code
-        uses: actions/checkout@v4
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-      - name: Display Python version
-        run: python -c "import sys; print(sys.version)"
-      - name: Decrypt parameters.py
-        shell: bash
-        run: .github/scripts/decrypt_parameters.sh
-        env:
-          PARAMETER_PASSWORD: ${{ secrets.PARAMETER_PASSWORD }}
-          CLOUD_PROVIDER: ${{ matrix.cloud-provider }}
-      - name: Download wheel(s)
-        uses: actions/download-artifact@v4
-        with:
-          name: wheel
-          path: dist
-      - name: Show wheels downloaded
-        run: ls -lh dist
-        shell: bash
-      - name: Upgrade setuptools, pip and wheel
-        run: python -m pip install -U setuptools pip wheel
-      - name: Install tox
-        run: python -m pip install tox
-      - if: ${{ contains('macos', matrix.os.download_name) }}
-        name: Run doctests
-        run: python -m tox -e "py${PYTHON_VERSION}-doctest-notudf-ci"
-        env:
-          PYTHON_VERSION: ${{ matrix.python-version }}
-          cloud_provider: ${{ matrix.cloud-provider }}
-          PYTEST_ADDOPTS: --color=yes --tb=short --disable_sql_simplifier
-          TOX_PARALLEL_NO_SPINNER: 1
-        shell: bash
-      - name: Run tests (excluding doctests)
+      # do not run other tests for macos on aws
+      - if: ${{ !(matrix.os == 'macos-latest' && matrix.cloud-provider == 'aws') }}
```

Comment:
We can just skip macos here right?

Reply:
You mean we don't need to have the check for aws now?

Reply:
I guess right now no macos should run in parallel in merge gate right?